File: Excel/500-599/559/559 Challenge.py
import pandas as pd

path = "559 Max of first N elements.xlsx"
input = pd.read_excel(path, usecols="A:D", skiprows=1)
test = pd.read_excel(path, usecols="F:I", skiprows=1, names=["A", "B", "C", "D"])

result = input.cummax()
result.columns = ["A", "B", "C", "D"]

print(result.equals(test)) # True

File: Excel/500-599/559/559 Challenge.R
library(tidyverse)
library(readxl)

path = "Excel/559 Max of first N elements.xlsx"
input = read_excel(path, range = "A2:D13")
test  = read_excel(path, range = "F2:I13")

output = input %>% 
  mutate(across(everything(), ~cummax(.))) 

all.equal(output, test, check.attributes = FALSE)
#> [1] TRUE